src/libraries/Native/Unix/System.Security.Cryptography.Native/opensslshim.c

```diff
@@ -116,8 +116,7 @@ static bool OpenLibrary()
     return libssl != NULL;
 }
 
-__attribute__((constructor))
-static void InitializeOpenSSLShim()
+void InitializeOpenSSLShim(void)
 {
     if (!OpenLibrary())
```

Comment:
This function now needs to ensure mutual exclusion because two different managed libraries could trip calls in parallel.

Specifically, first caller does work, subsequent callers have to block until the first caller finishes.

Reply:
Actually, this function looks relatively idempotent.  It might be OK.

Reply:
I think the mutual exclusion is the reason why we do not call this entry directly, but rely on triggering a static constructor, which is guaranteed to execute once.

Reply:
Well... it's guaranteed to execute once per type.  But since it's an internal type included in multiple assemblies (System.Security.Cryptography.Algorithms, System.Security.Cryptography.OpenSsl, System.Security.Cryptography.X509Certificates, and System.Net.Security) there are multiple cctors which can call it.

If all cctors execute on a single thread then mutex is guaranteed, but my understanding is they run on the invoking thread, so this has to account for 4-library parallel callers.

Reply:
And yes, I think it is idempotent. Shim init basically just binds pointers to implementing methods. Doing this twice should produce the same results.

Reply:
> since it's an internal type included in multiple assemblies

Oh right. That will call for every clone of the `CryptoInitializer`, which would be redundant, but ok thing to do.